conf.d/hydro.fish

```diff
@@ -75,8 +75,7 @@ function _hydro_prompt --on-event fish_prompt
 
         test -z \"\$$_hydro_git\" && set --universal $_hydro_git \"\$branch \"
 
-        ! command git diff-index --quiet HEAD 2>/dev/null ||
```

Comment:
I suspect removing the `git diff --quiet` check _may_ be a performance regression when multiple files are changed on larger repos.

Reply:
I randomly modified a few files on the nerd-fonts repository and tested them, but found no regression.

<img width="199" alt="image" src="https://github.com/jorgebucaran/hydro/assets/17561793/718d6fbb-d7bc-4ddc-a8b8-20b5df715fb1">

<img width="1470" alt="image" src="https://github.com/jorgebucaran/hydro/assets/17561793/4034e7f5-900e-42b3-bd6d-478ef75fde1f">

Reply:
I think that's a valid way to benchtest it.

Reply:
Difference is minimal too.

Reply:
Have you tested on a large repo, such as chromium, gecko-dev, or nixpkgs? nerd-fonts doesn't seem that big.

Reply:
there is indeed a performance loss🥲

nixpkgs repo:
<img width="1262" alt="image" src="https://github.com/jorgebucaran/hydro/assets/17561793/94c0edbc-660a-4d42-ba82-15587ebee7b7">

Reply:
more test:
<img width="921" alt="image" src="https://github.com/jorgebucaran/hydro/assets/17561793/3623e13a-3655-4f98-9107-648d9d977c59">

Reply:
Based on the above test results, we can also change it to:
```
count (git status --porcelain=v2 --untracked-files=no) >/dev/null ||
            count (command git ls-files --others --exclude-standard) >/dev/null && set info \"$hydro_symbol_git_dirty\"
```